cpu/cc3200/periph/timer.c

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (C) 2019 Ludwig Maximilian Universität
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup        cpu_cc3200
+ * @{
+ *
+ * @file
+ * @brief           Timer implementation
+ *
+ * @author          Wladislaw Meixner <wladislaw.meixner@campus.lmu.de>
+ */
+#include <stdlib.h>
+
+#include "panic.h"
+#include <sys/types.h>
+#include <thread.h>
+
+#include "periph/timer.h"
+#include "periph_conf.h"
+#include "xtimer.h"
+
+#include "vendor/hw_ints.h"
+#include "vendor/hw_memmap.h"
+#include "vendor/hw_timer.h"
+#include "vendor/hw_types.h"
+#include "vendor/rom.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#define LOAD_VALUE (0xffff)
+
+/**
+ * @brief Timer state memory
+ */
+static timer_isr_ctx_t isr_ctx[TIMER_NUMOF];
+
+/**
+ * @brief get timer control register by ID. Each register is 0x1000 apart
+ * starting at TIMERA0_BASE
+ *
+ *
+ */
+static inline cc3200_timer_t *timer(tim_t num)
+{
+    return (cc3200_timer_t *)(TIMERA0_BASE + (num << 12));
+}
+
+/**
+ * @brief returns the timer peripheral register used to enable or disable
+ * hardware peripheral.
+ *
+ */
+static inline cc3200_arcm_reg_t *timer_periph_reg(tim_t num)
+{
+    return (cc3200_arcm_reg_t *)((&ARCM->GPT_A0) +
+                                 sizeof(cc3200_arcm_reg_t) * num);
+}
+
+/**
+ * @brief timer interrupt handler
+ *
+ * @param[in] dev GPT instance number
+ */
+static void timer_irq_handler(tim_t dev)
+{
+    timer_clear(dev, 0);
+    isr_ctx[dev].cb(isr_ctx[dev].arg, 0); /* timer has one hw channel */
+    cortexm_isr_end();
+}
+
+#ifdef TIMER_0_EN
+void isr_timer0(void)
+{
+    timer_irq_handler(0);
+}
+#endif
+
+#ifdef TIMER_1_EN
+void isr_timer1(void)
+{
+    timer_irq_handler(1);
+}
+#endif
+
+#ifdef TIMER_2_EN
+void isr_timer2(void)
+{
+    timer_irq_handler(2);
+}
+#endif
+
+#ifdef TIMER_3_EN
+void isr_timer3(void)
+{
+    timer_irq_handler(3);
+}
+#endif
+
+/**
+ * @brief Get the irq handler object
+ *
+ * @param dev
+ * @return void*
+ */
+static inline void *get_irq_handler(tim_t dev)
+{
+    switch (dev) {
+#ifdef TIMER_0_EN
+    case 0:
+        return isr_timer0;
+#endif
+#ifdef TIMER_1_EN
+    case 1:
+        return isr_timer1;
+#endif
+#ifdef TIMER_2_EN
+    case 2:
+        return isr_timer2;
+#endif
+#ifdef TIMER_3_EN
+    case 3:
+        return isr_timer3;
+#endif
+    default:
+        /* requested irq handler for invalid timer */
+        DEBUG("REQUESTED IRQ FOR INVALID TIMER");
+        assert(0);
+    };
+}
+
+#define ROM_TimerConfigure \
+    ((void (*)(unsigned long ulBase, unsigned long ulConfig))ROM_TIMERTABLE[2])
```

Comment:
How does this macro work?

Reply:
For some reason I always got a compiler warning that ROM_TimerConfigure when compiling the timer. This simply maps to the address of the timer configure function inside the MCUs ROM.  
But I will try to remove the definition and see if it still compiles, maybe the problem has resolved itself :)

Reply:
If you want to keep it, please add a comment to explain what it does.

Reply:
After updating to RIOT master I could compile without this definition so I removed it.